egret/common/lazy_ptdf_utils.py

```diff
@@ -500,13 +542,16 @@ def add_violations(viol_lazy, int_viol_lazy, PFV, PFV_I, mb, md, solver, ptdf_op
 
     constr = mb.ineq_pf_branch_thermal_bounds
     viol_in_mb = mb._idx_monitored
+    pos_slacks = hasattr(mb, 'pf_slack_pos')
+    neg_slacks = hasattr(mb, 'pf_slack_neg')
```

Comment:
function arguments?

Reply:
I'm open to making this change, though I do think in lazy_ptdf_utils more indirection is called for. These functions can more easily be used on more models/blocks if they infer things about their state. As-is, this function gets called within the "solve_*" functions, not as part of a model build. Thoughts?

Reply:
That is a good point. I was definitely thinking about model building, not solving.